mlflow/store/azure_blob_artifact_repo.py

```diff
@@ -0,0 +1,114 @@
+import os
+import re
+
+from six.moves import urllib
+
+from mlflow.entities.file_info import FileInfo
+from mlflow.store.artifact_repo import ArtifactRepository
+from mlflow.utils.file_utils import build_path, get_relative_path, TempDir
+
+
+class AzureBlobArtifactRepository(ArtifactRepository):
+    """
+    Stores artifacts on Azure Blob Storage.
+
+    This repository is used with URIs of the form
+    ``wasbs://<container-name>@<ystorage-account-name>.blob.core.windows.net/<path>``,
```

Comment:
Btw, the Hadoop FS scheme is just [wasb://](https://hadoop.apache.org/docs/current/hadoop-azure/index.html). We may want to use that scheme instead.

Reply:
They also have wasbs:// for a TLS-secured connection, and recommend that by default: https://docs.microsoft.com/en-us/azure/hdinsight/hdinsight-hadoop-use-blob-storage#address-files-in-azure-storage. We actually also recommend that in the Azure DB docs.

Reply:
Wow, did not realize there is an "unencrypted" endpoint. Sounds good.